From c29bb5d613490922e9c4cec2b6c810ba495de114 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Mon, 4 May 2015 12:42:42 +0300
Subject: [PATCH 114/538] usb: host: xhci: mvebu: add reset on resume quirk

	The resume operation of mvebu xHCI host have some issues,
	so The XHCI_RESET_ON_RESUME quirk is added for it.

Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/18774
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>

Conflicts:
	drivers/usb/host/xhci-mvebu.c

fix compilation error
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>

Change-Id: I4261625f738086f698f028416b66074028d45bf4
Reviewed-on: http://vgitil04.il.marvell.com:8080/27490
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 drivers/usb/host/xhci-mvebu.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/usb/host/xhci-mvebu.c b/drivers/usb/host/xhci-mvebu.c
index 1eefc98..c84bf15 100644
--- a/drivers/usb/host/xhci-mvebu.c
+++ b/drivers/usb/host/xhci-mvebu.c
@@ -13,6 +13,7 @@
 #include <linux/platform_device.h>
 
 #include "xhci-mvebu.h"
+#include "xhci.h"
 
 #define USB3_MAX_WINDOWS	4
 #define USB3_WIN_CTRL(w)	(0x0 + ((w) * 8))
@@ -41,6 +42,14 @@ static void xhci_mvebu_mbus_config(void __iomem *base,
 	}
 }
 
+static void xhci_mvebu_quirks(struct platform_device *pdev)
+{
+	struct usb_hcd *hcd = platform_get_drvdata(pdev);
+	struct xhci_hcd	*xhci = hcd_to_xhci(hcd);
+
+	xhci->quirks |= XHCI_RESET_ON_RESUME;
+}
+
 int xhci_mvebu_mbus_init_quirk(struct platform_device *pdev)
 {
 	struct resource	*res;
@@ -68,5 +77,7 @@ int xhci_mvebu_mbus_init_quirk(struct platform_device *pdev)
 	 */
 	iounmap(base);
 
+	xhci_mvebu_quirks(pdev);
+
 	return 0;
 }
-- 
1.9.1

